scm/scm.go

```diff
@@ -77,10 +78,10 @@ type SCM interface {
 }
 
 // NewSCMClient returns a new provider client implementing the SCM interface.
-func NewSCMClient(logger *zap.SugaredLogger, provider, token string) (SCM, error) {
+func NewSCMClient(logger *zap.SugaredLogger, client *github.Client, provider, token string) (SCM, error) {
```

Comment:
We should avoid passing a GitHub-specific argument to this constructor. Instead we should create the `github.Client` inside this function or in the `NewGithubSCMClient` function. I think this would actually simplify things, making `AdminService` and `AutograderService` contain just the `SCM` interface instead of `scms.GithubApp` concrete type.